### PR TITLE
Fix autorestart feature (and test) and fix pid bug

### LINKED ISSUE #941: [core] processes run in foreground do not have PID files
collector status command seems to return an incorrect message. reported from a user, also tested locally:

$ dd-agent status
Log file is unwritable: '/var/log/datadog/collector.log'
2014-05-13 16:38:36,284 | DEBUG | dd.collector | config(config.py:523) | No proxy configured
2014-05-13 16:38:36,284 | DEBUG | dd.collector | util(util.py:387) | Using temporary pid file: /tmp/dd-agent.pid
2014-05-13 16:38:36,284 | INFO | dd.collector | daemon(daemon.py:250) | Agent is not running
Agent is not running

$ sudo supervisorctl -c /etc/dd-agent/supervisor.conf status
datadog-agent:collector          RUNNING    pid 6279, uptime 3 days, 21:34:41
datadog-agent:dogstatsd          RUNNING    pid 6277, uptime 3 days, 21:34:41
datadog-agent:forwarder          RUNNING    pid 6278, uptime 3 days, 21:34:41
datadog-agent:pup                RUNNING    pid 6280, uptime 3 days, 21:34:41

### PR DESCRIPTION
**Pull request by degemer**
Fixes #941

### CODE REVIEW

**Comment by degemer** [comment]
Added @LeoCavaille modifications to repair autorestart feature

File: daemon.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 import logging
 import errno
 import signal
+import traceback
```
Comment (LeoCavaille) [inline daemon.py]: Not needed :)!

**Comment by degemer** [comment]
After the modifications about the autorestart feature, testAutoRestart was still failing on Travis: it was because of an uncatched exception (fail to get CPU metrics) during the CPU check.

File: checks/system/unix.py [hunk 2]
```diff
@@ -757,149 +757,148 @@ def get_value(legend, data, name, filter_value=None):
...
+                        rel_size = [s/count for s in size]
+                        dot = lambda v1, v2: reduce(operator.add, map(operator.mul, v1, v2))
+                        return format_results(dot(user, rel_size),
+                                              dot(kern, rel_size),
+                                              dot(wait, rel_size),
+                                              dot(idle, rel_size),
+                                              0.0)
+            else:
+                    self.logger.warn("CPUStats: unsupported platform")
+                    return False
```
Comment (LeoCavaille) [inline checks/system/unix.py]: weird indentation in here ?!

**Comment by remh** [comment]
@LeoCavaille @degemer what should be the milestone for this PR ?

**Comment by LeoCavaille** [comment]
@remh I'd rather see that in 5.2, it's clearly a bad bug so let's fix it.

**Comment by LeoCavaille** [comment]
Looking good.

**Pull request merged**